open-craft/opencraft
# SE-1607 extra emails on build failure

This PR adds the extra monitoring email addresses as recipients to the email send when an appserver fails to provision. This allows us to specify extra addresses to receive alerts for the periodic builds. It is also general to allow using this for example to ping a PR author if the sandbox fails to provision.

This also includes the patch from #301 (link to instance from admin) and the complementary link from instance page to admin for convenience. As a result, this closes [SE-1409](https://tasks.opencraft.com/browse/SE-1409).

**Screenshots**:

![1571115836](https://user-images.githubusercontent.com/9714796/66801624-3addee80-ef61-11e9-91ce-7e762a6d09b3.png)

![1571114824](https://user-images.githubusercontent.com/9714796/66801625-3addee80-ef61-11e9-8eaf-09293d3b4244.png)


**Testing instructions**:

1. Add an email address you can monitor to an instance's `additional_monitoring_emails` field.
2. set the instance up to fail (eg. choose an invalid git revision for edx-platform, set extra settings that are known to fail, etc.)
3. launch an appserver
4. verify that the email entered in step 1 is emailed with details of the appserver failure when it moves to configuration failed state.
5. Navigate to the settings tab of an instance, find and click the edit icon next to 'Current configuration', verify that it navigates to the admin page for that instance  (see screenshots for how buttons should look)
6. on the admin page for the instance, find the 'view on site' button on the top right, and verify that clicking that navigates to the frontend page for that instance.

**Reviewers**
